src/main/scala/model/Bunny.scala

```diff
@@ -0,0 +1,142 @@
+package model
+
+import scala.util.Random
+
+/**
+ * Represents a Bunny.
+ */
+sealed trait Bunny{
+  val genotype: CompletedGenotype
+  val mom: Option[Bunny]
+  val dad: Option[Bunny]
+  val age: Int
+  val isAlive: Boolean
+
+  override def toString: String = {
+    super.toString + "\n" + genotype.genes
+      .map(g => "\t" + g._1 + ": "+ g._2.getVisibleTrait.toString.toLowerCase + " (" + g._2.getLetters + ")")
+      .reduce(_ + "\n" + _)
+      .replace("_", " ") + "\n"
+  }
+}
+
+/**
+ * Represents a Bunny which is still alive.
+ * @param genotype  contains the Bunny genes, it must be completed (contain all the available Genes of the world)
+ * @param mom       the bunny's mom
+ * @param dad       the bunny's dad
+ * @param age       how many generations the Bunny has spent in the world.
+ */
+case class AliveBunny(genotype: CompletedGenotype, mom:Option[Bunny], dad:Option[Bunny], age:Int) extends Bunny{
+  override val isAlive: Boolean = true
+}
+
+/**
+ * Represents a Bunny which dead.
+ * @param genotype  contains the Bunny genes, it must be completed (contain all the available Genes of the world)
+ * @param mom       the bunny's mom
+ * @param dad       the bunny's dad
+ * @param age       how many generations the Bunny has spent in the world.
+ */
+case class DeadBunny(genotype: CompletedGenotype, mom:Option[Bunny], dad:Option[Bunny], age:Int) extends Bunny{
+  override val isAlive: Boolean = false
+}
+
+/**
+ * Represents the abstraction of a Bunny that as just been created.
+ */
+abstract class InitBunny() extends Bunny{
+  override val age: Int = 0
+  override val isAlive: Boolean = true
+}
+
+/**
+ * Represents a Bunny that as just been created.
+ */
+case class ChildBunny(genotype: CompletedGenotype, mom:Option[Bunny], dad:Option[Bunny]) extends InitBunny
+
+/**
+ * Represents the first Bunny which appears in the world, so it does not have a mom and a dad.
+ */
+case class FirstBunny(genotype: CompletedGenotype) extends InitBunny {
+  override val mom: Option[Bunny] = Option.empty
+  override val dad: Option[Bunny] = Option.empty
+}
+
+object BunnyUtils {
+  val CHILDREN_NUMBER = 4
+
+  /**
+   * @return a FirstBunny with the "base" allele for each gene
+   */
+  def getBaseFirstBunny: FirstBunny =
+    FirstBunny(CompletedGenotype(Genes.values.unsorted.map(gk =>
+      (gk, StandardGene(gk, StandardAllele(gk.base), StandardAllele(gk.base)))).toMap))
+
+  /**
+   * @return a FirstBunny with a random allele for each gene
+   */
+  def getRandomFirstBunny: FirstBunny = {
```

Comment:
Ci serve o è solo a fini di test?

Reply:
Nella nostra idea di simulazione non lo usiamo, infatti a un certo punto l'avevo fatto nel file di test. Ho pensato che in generale potesse essere utile in altri contesti quindi dato che ormai era fatto ho pensato di lasciarlo qui disponibile a chiunque usi un Bunny.

Reply:
Per me al momento si può lasciare poi al massimo lo togliamo in seguito